baseframe/templates/baseframe/forms.html

```diff
@@ -247,4 +248,26 @@
     });
   </script>
   {%- endif %}
-{% endmacro %}
\ No newline at end of file
+{% endmacro %}
+
+{% macro add_map_marker_widget(field) %}
+  <style>
+    .map {
+      width: 100%;
+      height: 20em;
+      margin-top: 12px;
+    }
+  </style>
+  <input type="text" class="form-control" id="{{field.location_id}}" placeholder="Enter a location" size="70">
+  <div id="{{field.map_id}}" class='map'></div>
+  <script type="text/javascript">
+    document.addEventListener("DOMContentLoaded", function(event) {
+      new window.Baseframe.MapMarker({
+        map_id: "{{field.map_id}}",
+        longitude_id: "{{field.longitude_id}}",
+        latitude_id: "{{field.latitude_id}}",
+        location_id: "{{field.location_id}}"
```

Comment:
These three ids -- location, latitude and longitude -- should be auto-generated from the field's id. These are internal API detail anyway, the view doesn't need to know how the form works internally.